### ISSUE TITLE
Tensorflow in R

### ISSUE THREAD

**Issue by innosoft21** [innosoft21/workspace#1]
devtools::install_github("rstudio/keras")
library(keras)
install_keras()
iris <- iris
library(corrplot)
M <- cor(iris[,1:4])
corrplot(M, method="circle")
iris[,5] <- as.numeric(as.factor(unlist(iris[,5]))) -1
iris <- as.matrix(iris)
dimnames(iris) <- NULL
iris_x <- normalize(iris[,1:4])
iris_mat <- cbind(iris_x, iris[,5])
head(iris_mat)
ind <- sample(2, nrow(iris_mat), replace=TRUE, prob=c(0.67, 0.33))
iris.training <- iris_mat[ind==1, 1:4]
iris.test <- iris_mat[ind==2, 1:4]
iris.trainingtarget <- iris_mat[ind==1, 5]
iris.testtarget <- iris_mat[ind==2, 5]
iris.trainLabels <- to_categorical(iris.trainingtarget)
iris.testLabels <- to_categorical(iris.testtarget)
set.seed(777)
model <- keras_model_sequential()
model %>%
  layer_dense(units = 8, activation = 'relu', input_shape = c(4)) %>%
  layer_dense(units = 3, activation = 'softmax')
summary(model)
get_config(model)
get_layer(model, index = 1)
model$layers
model %>% compile(
  loss = 'categorical_crossentropy',
  optimizer = 'adam',
  metrics = 'accuracy'
)
model %>% fit(
  iris.training,
  iris.trainLabels,
  epochs = 500,
  batch_size = 5,
  validation_split = 0.1
)
classes <- model %>% predict_classes(iris.test, batch_size = 128)
table(iris.testtarget, classes)
score <- model %>% evaluate(iris.test, iris.testLabels, batch_size = 128)
print(score)
